From 6c71d8d16faf7c13b786f3f35fef6ab533bad2c1 Mon Sep 17 00:00:00 2001
From: Daniel Petry <dpetry@anaconda.com>
Date: Tue, 21 Jan 2025 17:45:23 -0600
Subject: [PATCH 07/14] Add conda prefix to inductor include & lib paths

Currently inductor doesn't look in conda's includes and libs. This results in
errors when it tries to compile, if system versions are being used of
dependencies (e.g., sleef).

Note that this is for inductor's JIT mode, not its AOT mode, for which the
end user provides a <filename>_compile_flags.json file.

Updated to use `sysconfig.get_config_vars("prefix")` per
https://github.com/conda-forge/pytorch-cpu-feedstock/issues/424
and https://github.com/conda-forge/pytorch-cpu-feedstock/issues/447.
---
 torch/_inductor/cpp_builder.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/torch/_inductor/cpp_builder.py b/torch/_inductor/cpp_builder.py
index e2cb445ed10..89905bec464 100644
--- a/torch/_inductor/cpp_builder.py
+++ b/torch/_inductor/cpp_builder.py
@@ -1410,10 +1410,12 @@ def get_cpp_torch_options(
         + python_include_dirs
         + torch_include_dirs
         + omp_include_dir_paths
+        + [sysconfig.get_config_vars("prefix") + '/include']
+        + [sysconfig.get_config_vars("prefix") + '/targets/@CUDA_TARGET@/include']
     )
     cflags = sys_libs_cflags + omp_cflags
     ldflags = omp_ldflags
-    libraries_dirs = python_libraries_dirs + torch_libraries_dirs + omp_lib_dir_paths
+    libraries_dirs = python_libraries_dirs + torch_libraries_dirs + omp_lib_dir_paths + [sys.prefix + '/targets/@CUDA_TARGET@/lib/stubs']
     libraries = torch_libraries + omp_lib
     passthrough_args = (
         sys_libs_passthrough_args + isa_ps_args_build_flags + omp_passthrough_args
